datafusion/core/src/dataframe/mod.rs

```diff
@@ -3019,11 +3019,11 @@ mod tests {
 
         assert_batches_sorted_eq!(
             [
-                "+-----+-----+----+-------+",
```

Comment:
Result of this test changes with this PR. I have analyzed the change, previously this tes was generating the following plan: 
```
    "ProjectionExec: expr=[c1@0 as c1, c2@1 as c2, c3@2 as c3, CAST(c2@1 AS Int8) + c3@2 as sum]",
    "  RepartitionExec: partitioning=RoundRobinBatch(8), input_partitions=1",
    "    SortExec: expr=[c1@0 ASC,c2@1 ASC,c3@2 ASC], preserve_partitioning=[false]",
    "      GlobalLimitExec: skip=0, fetch=1",
    "        CoalescePartitionsExec",
    "          CoalesceBatchesExec: target_batch_size=8192, fetch=1",
    "            FilterExec: c2@1 = 3 AND c1@0 = a",
    "              RepartitionExec: partitioning=RoundRobinBatch(8), input_partitions=1",
    "                CsvExec: file_groups={1 group: [[Users/akurmustafa/projects/synnada/datafusion-upstream/testing/data/csv/aggregate_test_100.csv]]}, projection=[c1, c2, c3], has_header=true",
```
After the changes in this PR, following plan is generated
```
    "  RepartitionExec: partitioning=RoundRobinBatch(8), input_partitions=1",
    "    SortExec: TopK(fetch=1), expr=[c3@2 ASC], preserve_partitioning=[false]",
    "      CoalescePartitionsExec",
    "        CoalesceBatchesExec: target_batch_size=8192",
    "          FilterExec: c2@1 = 3 AND c1@0 = a",
    "            RepartitionExec: partitioning=RoundRobinBatch(8), input_partitions=1",
    "              CsvExec: file_groups={1 group: [[Users/akurmustafa/projects/synnada/datafusion-upstream/testing/data/csv/aggregate_test_100.csv]]}, projection=[c1, c2, c3], has_header=true",
```
I think the plan is effectively same. However, the result of the test is not deterministic as is.